test/functional/terminal/parser_spec.lua

```diff
@@ -12,4 +16,36 @@ describe(':terminal', function()
     api.nvim_chan_send(chan, '\027]8;;https://example.com\027\\Example\027]8;;\027\n')
     assert_alive()
   end)
+
+  it('handles OSC-2 title setting', function()
+    -- OSC-2 should set title.
+    local chan = api.nvim_open_term(0, {})
+    local input = OSC_PREFIX .. '2;This title set with OSC 2' .. BEL
+    api.nvim_chan_send(chan, input)
+    --- @type string
+    local term_title = api.nvim_buf_get_var(0, 'term_title')
+    assert.Equal(term_title, 'This title set with OSC 2')
+    assert_alive()
+  end)
+
+  it('handles OSC-0 title and icon setting', function()
+    -- OSC-0 should set title and icon name to the same string. We currently ignore the icon name,
+    -- but the title should still be reflected.
+    local chan = api.nvim_open_term(0, {})
+    local input = OSC_PREFIX .. '0;This title set with OSC 0' .. BEL
+    api.nvim_chan_send(chan, input)
+    --- @type string
+    local term_title = api.nvim_buf_get_var(0, 'term_title')
+    assert.Equal(term_title, 'This title set with OSC 0')
+    assert_alive()
+  end)
+
+  it('handles control character following OSC prefix #34028', function()
+    local chan = api.nvim_open_term(0, {})
+    local input = OSC_PREFIX .. NUL .. '0;aaaaaaaaaaaaaaaaaaaaaaaaaaa\r\n'
+    --                          ^ Having a ctrl char following the OSC_PREFIX causes
+    --                            `string_fragment()` to be called while in OSC_COMMAND mode.
+    api.nvim_chan_send(chan, input)
+    assert_alive()
+  end)
```

Comment:
All three of these tests pass on master without your change, so I don't think these reproduce the issue you're describing.

Reply:
> So i'm wondering if you are on an apple silicon mac?

Indeed I am. Thanks for looking into this.